migration/handler.go

```diff
@@ -122,3 +122,77 @@ func (h *upgradeSchemaHandler) validate(ctx weave.Context, db weave.KVStore, tx
 
 	return &msg, nil
 }
+
+// SchemaRoutingHandler is a container that clubs toghether message handlers
+// for a single type message but different schema formats. Each handler is
+// registered together with the lowest schema version that it supports. For example
+//
+//   handler := SchemaRoutingHandler{
+//     1: &MyHandlerVersionAlpha{},
```

Comment:
Wow, I totally thought this was a map, then just saw it was an array. Yeah... I remember this notation, didn't know you could skip to make a sparse array.Nice idea.